sdk/messaging_servicebus/src/service_bus/client.rs

```diff
@@ -0,0 +1,122 @@
+use std::sync::Arc;
+
+use crate::{
+    service_bus::{
+        peek_lock_message, peek_lock_message2, receive_and_delete_message, send_message,
+        PeekLockResponse,
+    },
+    utils::body_bytes_to_utf8,
+};
+use chrono::Duration;
+use ring::hmac::Key;
+
+use azure_core::{error::Error, HttpClient};
+
+/// Client object that allows interaction with the ServiceBus API
+pub struct Client {
+    http_client: Arc<dyn HttpClient>,
+    namespace: String,
+    queue: String,
+    policy_name: String,
+    signing_key: Key,
+}
+
+impl Client {
+    /// Creates a new client instance
+    pub fn new<N, Q, P, K>(
+        http_client: Arc<dyn HttpClient>,
+        namespace: N,
+        queue: Q,
+        policy_name: P,
+        policy_key: K,
+    ) -> Result<Client, Error>
+    where
+        N: Into<String>,
+        Q: Into<String>,
+        P: Into<String>,
+        K: AsRef<str>,
+    {
+        let signing_key = Key::new(ring::hmac::HMAC_SHA256, policy_key.as_ref().as_bytes());
+
+        Ok(Client {
+            http_client,
+            namespace: namespace.into(),
+            queue: queue.into(),
+            policy_name: policy_name.into(),
+            signing_key,
+        })
+    }
+
+    /// Sends a message to the queue
+    pub async fn send_message(&mut self, msg: &str) -> Result<(), Error> {
```

Comment:
no need to make self mutable reference. 
```suggestion
    pub async fn send_message(&self, msg: &str) -> Result<(), Error> {
```

Reply:
done ✅